### PR TITLE
consider `@Input` annotations

### PR DESCRIPTION
**Pull request by t1**

### CODE REVIEW

File: client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java [hunk 1]
```diff
@@ -45,6 +47,10 @@ private String graphQlInputTypeName(TypeInfo type) {
     }
 
     private String simpleInputTypeName(TypeInfo type) {
+        if (type.isAnnotated(Input.class))
+            return type.getAnnotation(Input.class).value();
+        if (type.isAnnotated(Type.class))
```
Comment (phillip-kruger) [inline client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java:52]: Input should not use `@Type`. First look `@Input`, then `@Name` then `className`. On the server we also have a NamingStrategy when using ClassName. Default is just the ClassName, then you also MergeInnerClass (so Inner class will do parentClassNameInnerClassName, Or you can do Full then it use the full packageName. Naming Strategy can be set via config.

**Review by phillip-kruger — CHANGES_REQUESTED** [review]

File: client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java (hunk 1 above)
Comment (t1) [reply client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java]: You're absolutely right. I just fixed that.

File: client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java [hunk 2]
```diff
@@ -45,6 +46,8 @@ private String graphQlInputTypeName(TypeInfo type) {
     }
 
     private String simpleInputTypeName(TypeInfo type) {
+        if (type.isAnnotated(Input.class))
+            return type.getAnnotation(Input.class).value();
         if (type.isAnnotated(Name.class))
             return type.getAnnotation(Name.class).value() + "Input";
```
Comment (phillip-kruger) [inline client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java:52]: Only append 'Input' on classname. same as server

**Review by phillip-kruger — APPROVED** [review]

File: client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java (hunk 2 above)
Comment (t1) [reply client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java]: Okay... I'm not really convinced that this is the best way to go, but consistency with the server beats any argument ;-)

**Review by phillip-kruger — APPROVED** [review]
Thanks @t1!

**Pull request merged**